test/tests/LibStuffTest.cpp

```diff
@@ -633,4 +637,39 @@ struct LibStuff : tpunit::TestFixture {
         ASSERT_EQUAL(SFirstOfMonth(timeStamp4, -13), "2019-06-01");
         ASSERT_EQUAL(SFirstOfMonth(timeStamp4, -25), "2018-06-01");
     }
+
+    void SQResultTest() {
+        SQLite db(":memory:", 1000, 1000, 1);
+        db.beginTransaction(SQLite::TRANSACTION_TYPE::EXCLUSIVE);
+        db.write("CREATE TABLE testTable(id INTEGER PRIMARY KEY, name STRING, value STRING, created DATE);");
+        db.write ("INSERT INTO testTable VALUES(1, 'name1', 'value1', '2023-12-20');");
+        db.write ("INSERT INTO testTable VALUES(2, 'name2', 'value2', '2023-12-21');");
+        db.write ("INSERT INTO testTable VALUES(3, 'name3', 'value3', '2023-12-22');");
+        db.prepare();
+        db.commit();
+
+        db.beginTransaction(SQLite::TRANSACTION_TYPE::SHARED);
+        SQResult result;
+        db.read("SELECT name, value FROM testTable ORDER BY id;", result);
```

Comment:
I assume this will work correctly, but just in case, can you add an alias and test it works properly. ie: `select name as caca`